scvi/model/_utils.py

```diff
@@ -229,3 +230,32 @@ def _get_batch_code_from_category(
             batch_loc = np.where(batch_mappings == cat)[0][0]
             batch_code.append(batch_loc)
     return batch_code
+
+
+def _init_library_size(
+    adata: anndata.AnnData, n_batch: dict
+) -> Tuple[np.ndarray, np.ndarray]:
+    data = get_from_registry(adata, _CONSTANTS.X_KEY)
+    batch_indices = get_from_registry(adata, _CONSTANTS.BATCH_KEY)
+
+    library_log_means = np.zeros(n_batch)
+    library_log_vars = np.ones(n_batch)
+
+    for i_batch in np.unique(batch_indices):
```

Comment:
going to take a more extensive review later, but this needs to look at the anndata setup dictionary and iterate over all batch categories that are registered. There is a niche case with scarches that more categories are registered than are in anndata. The shapes of the `library_log_means`, var should be number of registered batches.

Reply:
I thought `n_batch` should be updated when the number of categories are extended in the archesmixin. The shapes are being set to `n_batch` in this case, and in the for loop we just update the indices present in the batch_indices and the leave the rest as defaults (0,1).

Reply:
ah this is correct